### ISSUE TITLE
False positive `used-before-assignment` with try/else

### ISSUE THREAD

**Issue by mattclay** [PyCQA/pylint#6790]
### Bug description

The `used-before-assignment` check reports a false positive with try/else starting with pylint 2.13.0.

```python
"""bug.py"""


def demo():
    """Demonstrate used-before-assignment regression in pylint 2.13.0 and later."""
    try:
        pass
    except ValueError:
        error = True
    else:
        return

    print(error)
```

### Command used

```shell
pylint bug.py
```

### Pylint output

```shell
************* Module bug
bug.py:13:10: E0601: Using variable 'error' before assignment (used-before-assignment)
```

### Expected behavior

No errors reported, as was the case for 2.12.2.

### Pylint version

```shell
pylint 2.14.0
astroid 2.11.5
Python 3.9.5 (default, Nov 23 2021, 15:27:38) 
[GCC 9.3.0]
```

### OS / Environment

Ubuntu 22.04.4

**Comment by jacobtylerwalls** [comment]
Thanks for the well-written report. I have a diff for this that if all goes well we should be able to backport to 2.14.1.

**Comment by mattclay** [comment]
@jacobtylerwalls Thanks for responding so quickly. I've opened a related issue (https://github.com/PyCQA/pylint/issues/6804) which involves `continue` instead of `return`, in case you want to treat them as separate issues.
